File: src/cz/encircled/eplayer/service/FolderScanService.java
package cz.encircled.eplayer.service;

import java.util.List;

/**
 * Created by Encircled on 11/06/2014.
 */
public interface FolderScanService {

    void start();

    void stop();

    FolderScanService addAllIfAbsent(List<String> absolutePaths);

    boolean removeFolder(String absolutePath);

    boolean addIfAbsent(String absolutePath);

}

File: src/main/java/cz/encircled/eplayer/service/FolderScanService.kt
package cz.encircled.eplayer.service

import cz.encircled.eplayer.model.PlayableMedia

/**
 * @author Encircled on 11/06/2014.
 */
interface FolderScanService {

    fun getMediaInFolder(path: String, callback: CancelableExecution<List<PlayableMedia>>)

}